src/Storefront/Controller/ProductController.php

```diff
@@ -157,11 +168,29 @@ public function saveReview(string $productId, RequestDataBag $data, SalesChannel
      */
     public function loadReviews(Request $request, RequestDataBag $data, SalesChannelContext $context): Response
     {
+        $this->checkReviewsActive($context);
+
         $reviews = $this->productReviewLoader->load($request, $context);
 
         return $this->renderStorefront('storefront/page/product-detail/review/review.html.twig', [
             'reviews' => $reviews,
             'ratingSuccess' => $request->get('success'),
         ]);
     }
+
+    /**
+     * @throws ReviewNotActiveExeption
+     */
+    protected function checkReviewsActive(SalesChannelContext $context): void
+    {
+        $showReview = $this->systemConfigService->get('core.listing.showReview', $context->getSalesChannel()->getId());
+
+        if (
```

Comment:
check for false and throw the exception then.

Reply:
We have write this function similar to `denyAccessUnlessLoggedIn()` and there is the same if and return struct